### DISCUSSION TITLE
Squash multiple commits into one commit from branch

### DISCUSSION THREAD

**Discussion by TrongHieu90** [gitextensions/gitextensions#9731] [Q&A]
Hi,

Ive seen there is an option to to create a squash commit but not sure it is what i want to do.
I want to combine multiple commits into one single commit then merge that commit back into master branch.
But from a squash commit I do not see any option to choose how many commits to squash.

**Comment by mstv** [comment]
You could do an Interactive Rebase.

![image](https://user-images.githubusercontent.com/36601201/141677004-82b94214-13ef-498f-a48d-f5bd15c2cced.png)

**Comment by gerhardol** [comment]
https://github.com/gitextensions/gitextensions/wiki/How-To:-Squash-and-Rebase-your-changes
